File: csharp/listdir/listdir.cs
/* 
  List directory contents
  2021 (c) Dmitry Stefankov 
*/

using System;
using System.IO;


class Program
{
        public static long DirsCount = 0;
        public static long FilesCount = 0;
        public static long TotalSize = 0;

        public static void DirectorySearch(string dir)
        {
          Console.WriteLine(dir);
          DirsCount++;
          try
          {
            foreach (string f in Directory.GetFiles(dir))
            {
               string FullFilename = dir+"\\"+Path.GetFileName(f);
               FileInfo fileInfo = new FileInfo(FullFilename);
               //FileAttributes attributes = fileInfo.Attributes;
               DateTime lastWriteTime = fileInfo.LastWriteTime;
               long FileSize = fileInfo.Length;
               FilesCount++;
               TotalSize += FileSize;

               //Console.WriteLine("  filename=\"{0}\\{1}\"",dir,Path.GetFileName(f));
               Console.WriteLine("datetime=\"{0}\"  filesize=\"{1}\"  filename=\"{2}\"",
                                  lastWriteTime.ToString("yyyy-MM-dd HH:mm:ss"), 
                                  FileSize.ToString(), FullFilename);
            }
            foreach (string d in Directory.GetDirectories(dir))
            {
              //Console.WriteLine(Path.GetFileName(d));
              DirectorySearch(d);
            }
          }
          catch (System.Exception ex)
          {
            Console.WriteLine(ex.Message);
          }
        }

        static int Main(string[] args)
        {
            // Display the number of command line arguments:
            // String ProgramFilename = args[0];
            //Console.WriteLine();
            // Invoke this program with an arbitrary set of command line arguments.
            string[] arguments = Environment.GetCommandLineArgs();
            //Console.WriteLine("GetCommandLineArgs: {0}", string.Join(", ", arguments));

            //System.Console.WriteLine(args.Length);
            if (args.Length < 1) {
              Console.WriteLine("List Directory, version 1.0");
              Console.WriteLine("Copyright (C) 2021 Dmitry Stefankov. All Rights Reserved.");
              Console.WriteLine("Usage : listdir <Dirname>");
              return 1;
            }

            string myDir = args[0];
            DirectorySearch(myDir);

            Console.WriteLine("Dirs=\"{0}\", Files=\"{1}\", TotalSize=\"{2}\"",
                              DirsCount.ToString(),FilesCount.ToString(),TotalSize.ToString());

            return 0;
        }
}

File: java/listdir/listdir.java
//
// List contents of directory
// Copyright (C) 2021 Dmitry Stefankov
//

import java.io.*;
import java.util.List;
import java.util.Date;
import java.nio.charset.Charset;
import java.nio.file.*;

public class listdir {


  public static void ListDir (String dirname) throws IOException {

    File dir=new File(dirname);
    File files[]=dir.listFiles();//files array stores the list of files

    for(int i=0;i<files.length;i++)
    {
        if(files[i].isFile()) //check whether files[i] is file or directory
        {
            String FullFileName = dirname+"\\"+files[i].getName();
            File file = new File(FullFileName);
            long file_size = file.length();
            long t = file.lastModified();
            Date d = new Date(t);
            System.out.println("  datetime=\""+d.toString()+"\""+"  size=\""+file_size+"\""+"  filename=\""+files[i].getName()+"\"");
            //System.out.println();

        }
        else if(files[i].isDirectory())
        {
            //System.out.println("Directory::"+files[i].getName());
            System.out.println(dirname+"\\"+files[i].getName());
            //System.out.println();
            ListDir(files[i].getAbsolutePath());
        }
    }
  }

  public static void main(String[] args) throws IOException {
     //System.out.println("There are " + args.length + " arguments given.");
     //for(int i = 0; i < args.length; i++) 
    //    System.out.println("The argument #" + (i+1) + " is " + args[i] + " and is at index " + i);
     if (args.length < 1) {
       System.out.println("List Directory Files, version 1.0");
       System.out.println("Copyright (c) 2021, Dmitry Stefankov");
       System.out.println("Usage: listdir.exe dirname");
       return;
     }
 
     String MainDir = args[0];
     ListDir(MainDir);
  }

}
